spark/src/main/scala/geotrellis/spark/reproject/TileRDDReproject.scala

```diff
@@ -81,6 +82,76 @@ object TileRDDReproject {
           }
       }
 
+    val (sampleKey, BufferedTile(sampleTile, sampleGridBounds)) = bufferedTiles.first() // XXX
```

Comment:
I think I see what you're saying. You should be able to avoid this by just using `SpatialKey` and performing the reproject on that. You'll get the `cellSize`, `Extent`, and `KeyBounds[SpatialKey]`. The last can be set on the `layer.metadata` using [`KeyBounds.setSpatialComponent`](https://github.com/locationtech/geotrellis/blob/master/spark/src/main/scala/geotrellis/spark/KeyBounds.scala#L140). That makes sense, since reprojection is spatial only operation. 

Tile dimensions are in the `metadata.layout` has the tile size, like you're using it. Does that satisfy the question about tile size?